### PR TITLE
Add depth of field effects

### PR DESCRIPTION
**Pull request by xen-000**
This PR adds the depth of field effects and entity from Alien Swarm, along with a new bloom glow effect.

Depth of field, controllable with env_dof_controller:
hl2_2020-09-21_15-04-41
[Also allows for focusing on a specific entity](https://www.youtube.com/watch?v=0v3BXNK1FYI)

Old bloom glow:
hl2_2020-09-10_23-47-17

New bloom glow:
hl2_2020-09-10_23-47-37

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/game/client/c_env_dof_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
+//====== Copyright � 1996-2005, Valve Corporation, All rights reserved. =======
+//
+// Purpose: 
+//
+//=============================================================================
```
Comment (Gocnak) [inline mp/src/game/client/c_env_dof_controller.cpp:5]: Remove empty comments and Purpose blocks

File: mp/src/game/client/viewpostprocess.cpp [hunk 2]
```diff
@@ -35,6 +35,15 @@ float g_flCustomAutoExposureMax = 0;
...
 float g_flCustomBloomScaleMinimum = 0.0f;
 
+// mapmaker controlled depth of field
+bool  g_bDOFEnabled = false;
+float g_flDOFNearBlurDepth = 50.0f;
+float g_flDOFNearFocusDepth = 200.0f;
+float g_flDOFFarFocusDepth = 250.0f;
+float g_flDOFFarBlurDepth = 1000.0f;
+float g_flDOFNearBlurRadius = 0.0f;
+float g_flDOFFarBlurRadius = 5.0f;
```
Comment (Gocnak) [inline mp/src/game/client/viewpostprocess.cpp:45]: Should these match the defaults in the fgd?

File: mp/src/game/client/viewpostprocess.cpp [hunk 3]
```diff
@@ -1609,6 +1618,43 @@ void DumpTGAofRenderTarget( const int width, const int height, const char *pFile
...
+	{
+		pbloomexpvar->SetFloatValue( g_flBloomExponent );
+	}
+
+	IMaterialVar *pbloomsaturationvar = downsample_mat->FindVar( "$bloomsaturation", &bFound, false );
+	if ( bFound )
+	{
+		pbloomsaturationvar->SetFloatValue( g_flBloomSaturation );
+	}
+	*/
```
Comment (Gocnak) [inline mp/src/game/client/viewpostprocess.cpp:1647]: Unnecessary code?

File: mp/src/game/client/viewpostprocess.cpp [hunk 4]
```diff
@@ -1609,6 +1618,43 @@ void DumpTGAofRenderTarget( const int width, const int height, const char *pFile
...
+
+	// downsample fb to rt0
+	SetRenderTargetAndViewPort( pDest );
+	// note the -2's below. Thats because we are downsampling on each axis and the shader
+	// accesses pixels on both sides of the source coord
+	pRenderContext->DrawScreenSpaceRectangle(	downsample_mat, 0, 0, nSrcWidth/4, nSrcHeight/4,
+												0, 0, nSrcWidth-2, nSrcHeight-2,
+												nSrcWidth, nSrcHeight );
```
Comment (Gocnak) [inline mp/src/game/client/viewpostprocess.cpp:1655]: Spacing

File: mp/src/game/client/viewpostprocess.cpp [hunk 5]
```diff
@@ -3086,4 +3132,195 @@ void DoImageSpaceMotionBlur( const CViewSetup &viewSetup )
...
+//=====================================================================================================================
+ConVar mat_dof_enabled( "mat_dof_enabled", "1" );
+ConVar mat_dof_override( "mat_dof_override", "0" );
+ConVar mat_dof_near_blur_depth( "mat_dof_near_blur_depth", "20.0" );
+ConVar mat_dof_near_focus_depth( "mat_dof_near_focus_depth", "100.0" );
+ConVar mat_dof_far_focus_depth( "mat_dof_far_focus_depth", "250.0" );
+ConVar mat_dof_far_blur_depth( "mat_dof_far_blur_depth", "1000.0" );
+ConVar mat_dof_near_blur_radius( "mat_dof_near_blur_radius", "10.0" );
+ConVar mat_dof_far_blur_radius( "mat_dof_far_blur_radius", "5.0" );
+ConVar mat_dof_quality( "mat_dof_quality", "3" );
```
Comment (Gocnak) [inline mp/src/game/client/viewpostprocess.cpp:3148]: Should these match the defaults in the fgd?

File: mp/src/game/server/env_dof_controller.cpp [hunk 6]
```diff
@@ -0,0 +1,199 @@
+//====== Copyright � 1996-2004, Valve Corporation, All rights reserved. =======
+//
+// Purpose: 
+//
+//=============================================================================
+
+#include "cbase.h"
+#include "baseentity.h"
+#include "entityoutput.h"
+//#include "convar.h"
```
Comment (Gocnak) [inline mp/src/game/server/env_dof_controller.cpp:10]: Remove unnecessary comments

File: mp/src/game/server/env_dof_controller.cpp [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+	SendPropFloat( SENDINFO(m_flFarFocusDepth), 0, SPROP_NOSCALE),
+	SendPropFloat( SENDINFO(m_flFarBlurDepth), 0, SPROP_NOSCALE),
+	SendPropFloat( SENDINFO(m_flNearBlurRadius), 0, SPROP_NOSCALE),
+	SendPropFloat( SENDINFO(m_flFarBlurRadius), 0, SPROP_NOSCALE),
+END_SEND_TABLE()
+
+//-----------------------------------------------------------------------------
+// Purpose: 
+//-----------------------------------------------------------------------------
+void CEnvDOFController::Spawn( void )
```
Comment (Gocnak) [inline mp/src/game/server/env_dof_controller.cpp:60]: Remove (void) parameters for newly added code.

File: mp/src/game/server/env_dof_controller.cpp [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+	m_flNearFocusDepth = setting.flNearFocusDistance;
+	
+	m_flFarBlurDepth = setting.flFarBlurDepth;
+	m_flFarBlurRadius = setting.flFarBlurRadius;
+	m_flFarFocusDepth = setting.flFarFocusDistance;
+
+	m_bDOFEnabled = ( m_flNearBlurRadius > 0.0f ) || ( m_flFarBlurRadius > 0.0f );
+}
+
+const float BLUR_DEPTH = 500.0f;
```
Comment (Gocnak) [inline mp/src/game/server/env_dof_controller.cpp:158]: Make this a `#define`

File: mp/src/game/server/env_dof_controller.h [hunk 9]
```diff
@@ -0,0 +1,64 @@
+
+#if !defined ( ENV_DOF_CONTROLLER_H )
+#define ENV_DOF_CONTROLLER_H
+#ifdef _WIN32
+#pragma once
+#endif
```
Comment (Gocnak) [inline mp/src/game/server/env_dof_controller.h:6]: Remove unnecessary header guards and just do `#pragma once`

File: mp/src/materialsystem/stdshaders/depthoffield_dx9.cpp [hunk 10]
```diff
@@ -200,7 +201,7 @@ BEGIN_VS_SHADER_FLAGS( DepthOfField_dx9, "Depth of Field", SHADER_NOT_EDITABLE )
 			vConst[9] = params[FARPLANE]->GetFloatValue();
 			
 			// 8192 is the magic number for HDR mode 3 (see FLOAT_RENDERPARM_DEST_ALPHA_DEPTH_SCALE in shaderapidx8.cpp)
```
Comment (Gocnak) [inline mp/src/materialsystem/stdshaders/depthoffield_dx9.cpp:203]: This comment no longer makes sense for the code, should be removed

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/src/game/client/viewpostprocess.cpp (hunk 4 above)
Comment (xen-000) [reply mp/src/game/client/viewpostprocess.cpp:1655]: All the draw calls in that file (and others) are spaced like that, so I believe this should be fixed in a separate formatting PR instead.

**Review by Gocnak — DISMISSED** [review]

**Review by Gocnak — APPROVED** [review]

**Pull request merged**